src/scripts/main.js

```diff
@@ -1,3 +1,21 @@
 'use strict';
 
-// write code here
+function parseSalary(salary) {
+  return parseInt(salary.replace('$', '').replace(',', ''));
+}
+
+function sortList(list) {
+  const items = Array.from(list.children);
+
+  items.sort((a, b) => {
+    const salaryA = parseSalary(a.getAttribute('data-salary'));
+    const salaryB = parseSalary(b.getAttribute('data-salary'));
+    return salaryB - salaryA;
+  });
+
+  // Re-append the sorted items back to the list
+  items.forEach(item => list.appendChild(item));
+}
+
+const employeeList = document.querySelector('ul');
```

Comment:
Ensure that the `employeeList` element exists in the DOM before calling `sortList`. If `document.querySelector('ul')` returns `null`, it will cause an error when passed to `sortList`.